logger/logger.go

```diff
@@ -149,40 +72,49 @@ func CreateLogger(options Options, key string) (*Logger, error) {
 		return nil, err
 	}
 
+	options.setDefaults()
 	logger := Logger{
 		Key:      key,
 		Messages: make(chan Message),
 		Options:  options,
 	}
-	go logger.processRequest()
+	go logger.handleMessages()
 	return &logger, nil
 }
 
 // Log sends a provided log message to LogDNA.
 func (logger *Logger) Log(message string) {
 	loggerMessage := Message{
-		Body: message,
+		Body:    message,
+		Options: logger.Options,
 	}
 	logger.Messages <- loggerMessage
 }
 
-// LogWithLevel sends a log message to LogDNA with a parametrized level.
-func (logger *Logger) LogWithLevel(message string, level string) error {
-	options := Options{Level: level}
-	err := options.validate()
+// LogWithOptions allows the user to update options uniquely for a given log message
+// before sending the log to LogDNA.
+func (logger *Logger) LogWithOptions(message string, options Options) error {
+	msgopts := logger.Options.merge(options)
```

Comment:
Could you explain how merge gets used here?

Reply:
this mainly just simplifies all of the handling for `Options` when sending lines. instead, now the message-level options are merged with the Logger's options, and `Message.Options` will _always_ contain everything we need to send a payload including things like `IngestURL`

the [previous code](https://github.com/logdna/logdna-go/pull/10/files#diff-ed0770fdbf87b0c6d536e33a99a8df9cL261-L265) also only worked when you set a `Hostname` on the message-level options